### PR TITLE
[KOGITO-186] - Allow jvm based executable on runtime Quarkus image

### PR DESCRIPTION
**Pull request by spolti**
Signed-off-by: Filippe Spolti <fspolti@redhat.com>

### CODE REVIEW

**Comment by spolti** [comment]
split the images into two different will be better to keep the native image smaller.

File: s2i/kogito-quarkus-ubi8-s2i-overrides.yaml [hunk 1]
```diff
@@ -11,6 +11,11 @@ labels:
 - name: "io.openshift.tags"
   value: "builder,kogito,quarkus"
 
+envs:
+- name: NONNATIVE
```
Comment (ricardozanini) [inline s2i/kogito-quarkus-ubi8-s2i-overrides.yaml]: Shouldn't be `NATIVE`?

File: s2i/modules/kogito-data-index/module.yaml [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
 name: kogito-data-index
-version: "0.3.0"
+version: "0.4.0"
 
 artifacts:
-- name: data-index-service-0.3.0-runner.jar
-  url: https://repo.maven.apache.org/maven2/org/kie/kogito/data-index-service/0.3.0/data-index-service-0.3.0-runner.jar
+- name: data-index-service-0.4.0-runner.jar
+  url: https://repo.maven.apache.org/maven2/org/kie/kogito/data-index-service/0.4.0/data-index-service-0.4.0-runner.jar
   md5: 12126dee53783fe09279f46f299611a1
```
Comment (ricardozanini) [inline s2i/modules/kogito-data-index/module.yaml]: I guess this build won't work unless we have the jar file in the repo. Let's wait until 0.4.0 release to change this line.

File: s2i/modules/kogito-quarkus-ubi8-s2i/README.md [hunk 3]
```diff
@@ -4,6 +4,9 @@ This S2I is considered image builder that is equipped with build tools
 
 - GraalVM
 - Apache Maven
+- JRE
+
+To perform non native builds set the env **NONNATIVE** to **true**.
```
Comment (ricardozanini) [inline s2i/modules/kogito-quarkus-ubi8-s2i/README.md]: `NATIVE`?

File: s2i/tests/features/kogito-quarkus-index.feature [hunk 4]
```diff
@@ -14,7 +14,7 @@ Feature: Kogito-quarkus-ubi8 feature.
 
   Scenario: verify if the binary index is available on /home/kogito
     When container is started with command bash
-    Then run sh -c 'ls /home/kogito/bin/data-index-service-*-runner.jar' in container and immediately check its output for /home/kogito/bin/data-index-service-0.3.0-runner.jar
+    Then run sh -c 'ls /home/kogito/bin/data-index-service-*-runner.jar' in container and immediately check its output for /home/kogito/bin/data-index-service-0.4.0-runner.jar
```
Comment (ricardozanini) [inline s2i/tests/features/kogito-quarkus-index.feature]: I guess this scenario will fail also. Can't we just look for a file starting with `/home/kogito/bin/data-index-service`?

**Review by ricardozanini — CHANGES_REQUESTED** [review]
Looks good! Many thanks for helping with this! Now I can safely test on my side with Operators and CLI

File: s2i/modules/kogito-data-index/module.yaml (hunk 2 above)
Comment (spolti) [reply s2i/modules/kogito-data-index/module.yaml]: good catch, I replaced everything from 0.3.0 to 0.4.0, will undo this.

File: s2i/kogito-quarkus-ubi8-s2i-overrides.yaml (hunk 1 above)
Comment (spolti) [reply s2i/kogito-quarkus-ubi8-s2i-overrides.yaml]: correct.

File: s2i/modules/kogito-quarkus-ubi8-s2i/README.md (hunk 3 above)
Comment (spolti) [reply s2i/modules/kogito-quarkus-ubi8-s2i/README.md]: yes.

File: s2i/tests/features/kogito-quarkus-index.feature (hunk 4 above)
Comment (spolti) [reply s2i/tests/features/kogito-quarkus-index.feature]: the module is copying the jar file as  data-index-service-0.3.0-runner.jar

**Review by ricardozanini — APPROVED** [review]

**Pull request merged**